### PR TITLE
Generic Layering and forceMove fixes

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Adjusts a few layers to use the proper layer defines, while also adjust a few planes.

Just a few generic fixes.

Also ensures equipping/unequipping items and moving properly calls `forceMove` instead of setting `loc`.

Fixes dead mobs utilizing a weird `Move`, `forceMove`, and `Moved` proc, allowing them to trigger a plethora of things they shouldn't

Brings back the squeak component on mice, as that's a better way to handle it for them, especially since the ghost mobs triggering it has been fixed

:cl: Fox McCloud
fix: Fixes weird behavior with ghosts being able to trigger a few things by crossing over them
/:cl:

### CODE REVIEW

File: code/datums/mutable_appearance.dm [hunk 1]
```diff
@@ -4,6 +4,11 @@
 
 // Mutable appearances are children of images, just so you know.
 
+/mutable_appearance/New()
+	..()
+	plane = FLOAT_PLANE // No clue why this is 0 by default yet images are on FLOAT_PLANE
```
Comment (AffectedArc07) [inline code/datums/mutable_appearance.dm]: https://www.floatplane.com/

Comment (taukausanake) [inline code/datums/mutable_appearance.dm]: Obligatory "we all float down here" comment

**Comment by Fox-McCloud** [comment]
Yeah, further testing shows this doesn't fix it. This problem is super finicky.

**Comment by Spacemanspark** [comment]
Been experiencing this for a while.

**Comment by Shadeykins** [comment]
I can confirm that I get this problem quite frequently. Changing the 'facing' of the mob in question usually seems to fix it (IE: turning from N to W).

**Comment by Bookdrizzt** [comment]
Yeah I experience this too.

**Comment by Fox-McCloud** [comment]
PR scope changed; this is just a generic layering+plane adjustment PR, along with a few movement/forcemove fixes.

The issue with invisible sprites persists.

**Comment by Fox-McCloud** [comment]
Well. Turns out the icon disappearing isn't our fault.

It's a BYOND issue: https://github.com/tgstation/tgstation/issues/36936

**Pull request merged**